src/main/java/service/ReportImpl.java

```diff
@@ -0,0 +1,37 @@
+package service;
+
+import dao.Reader;
+import dao.ReaderImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReportImpl implements Report {
+
+    private static final String FILE_PATH = "src/main/resources/input.csv";
+
+    @Override
+    public Map<String, Integer> report() {
+        Reader reader = new ReaderImpl();
+        List<String> data = reader.reader(FILE_PATH);
+        Map<String, Integer> fruitBalance = new HashMap<>();
+        for (String element : data) {
```

Comment:
this algorithm violites OCP from SOLID, use here strategy pattern instead: create separate handler for each operation and initialize Map<String, OperationHandler> with key as operation name and value as handler. You should have similar example in videos